mlflow/types/schema.py

```diff
@@ -685,16 +685,23 @@ def __repr__(self) -> str:
     def from_json_dict(cls, **kwargs):
         """
         Deserialize from a json loaded dictionary.
-        The dictionary is expected to contain `name`, `dtype` and `default` keys.
+        The dictionary is expected to contain `name`, `type` and `default` keys.
         """
-        if not {"name", "dtype", "default"} <= set(kwargs.keys()):
+        # For backward compatibility, we accept both `type` and `dtype` keys
+        if not {"name", "dtype", "default"} <= set(kwargs.keys()) and not {
+            "name",
+            "type",
+            "default",
+        } <= set(kwargs.keys()):
```

Comment:
/ai is it possible to simplify this condition?

Reply:
@serena-ruan Can we make this change?

Reply:
@harupy Do you have other comments? I think this https://github.com/mlflow/mlflow/pull/9565#discussion_r1319556184 seems even more complicated

Reply:
I prefers calling `issubet` which is more clear. :)